### DISCUSSION TITLE
App crashing on iOS simulator

### DISCUSSION THREAD

**Discussion by SandeepReddyPinniti** [dotnet/maui#21715] [Q&A]
Hi Team,

I have a basic login page, I'm running the app on iOS simualtor and developing the code in VS2022 on windows machine. The App crashes as soon as I focus the mouse on the Email text box. I have registered Unhandled Exception event, but the event is not firing. Will you please guide me in resolving this issue.

C#

Copy
 AppDomain.CurrentDomain.UnhandledException += HandleUnhandledException;

private void HandleUnhandledException(object sender, UnhandledExceptionEventArgs e)
{
    // Log or handle the exception here
    Exception ex = (Exception)e.ExceptionObject;
    HandleError(ex, "App.HandleUnhandledException");
}

XAML

Copy
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:viewModels="using:Biome.ViewModels"
             x:Class="Biome.LoginPage"
    x:DataType="viewModels:LoginViewModel">
    <ContentPage.BindingContext>
        <viewModels:LoginViewModel />
    </ContentPage.BindingContext>
    <VerticalStackLayout Padding="30,0" Spacing="25">
        <Image Source="biome.png" HeightRequest="185" Aspect="AspectFit" SemanticProperties.Description="biome" />
        <Grid HorizontalOptions="Center" VerticalOptions="Center">
            <Grid.RowDefinitions>
                <RowDefinition Height="45"></RowDefinition>
                <RowDefinition Height="5"></RowDefinition>
                <RowDefinition Height="45"></RowDefinition>
                <RowDefinition Height="5"></RowDefinition>
                <RowDefinition Height="45"></RowDefinition>
            </Grid.RowDefinitions>
            <Grid.ColumnDefinitions>
                <ColumnDefinition Width="500"></ColumnDefinition>
            </Grid.ColumnDefinitions>
            <Entry Grid.Row="0" Placeholder="Email Address" Text="{Binding LoginModel.Email}" />
            <Entry Grid.Row="2" Placeholder="Password" IsPassword="True" Text="{Binding LoginModel.Password}" />
            <Button Grid.Row="4" Text="Login" Command="{Binding LoginCommand}" />
        </Grid>
    </VerticalStackLayout>

</ContentPage>

Thanks

**Comment by ivanpovazan** [comment]
@SandeepReddyPinniti 
If you are using Xcode 15.3, and with it iOS 17.4 simulator, then you are most probably hitting this issue: https://github.com/dotnet/runtime/issues/98941 that prevents MAUI from officially supporting Xcode 15.3 which is tracked here: https://github.com/dotnet/maui/issues/21057

The fix will be released in the next servicing release.
Before then, as a workaround, you could try using an older Xcode version (15.2) or try a workaround listed in the description of the issue https://github.com/dotnet/runtime/issues/98941

**Reply by SandeepReddyPinniti** [reply]
Thanks for your response. I don't see an option install a specific version of xCode on Mac. Is there a way for it?

**Reply by ivanpovazan** [reply]
Yes, it is possible and you can have different versions installed at the same time, but it requires a few manual steps.
If you are not that familiar with command-line tooling on your Mac and you find it uncomfortable to do the following steps, I would not recommend doing them - to avoid bringing your system in an unwanted state. Nevertheless, here are the required steps. 

To download/install an older version of Xcode (e.g., 15.2) you would have to:
1. Manually download the installer by going to: https://developer.apple.com/download/all/?q=xcode and looking up Xcode 15.2 
2. Once it is downloaded, unpack/install it by opening the `.xip` file
3. Once the process is completed, `Xcode.app` will appear next to the `.xip` file
4. Rename the `Xcode.app` into something like: `Xcode_15.2.0.app` (this step is required as your system-wide version of Xcode, the current version you are using, has the same name in `Applications` folder)
5. Move the `Xcode_15.2.0.app` into your `Applications` (after this step your applications folder would look something like this:
```
/Applications/Xcode.app
/Applications/Xcode_15.2.0.app
```

After the above steps you will have everything set up, the only thing left to do is to select the default version of Xcode command-line tools to be used by your system. To do that, there are couple of additional steps required to switch back-and-forth between installed versions (as described in: https://developer.apple.com/library/archive/technotes/tn2339/_index.html) 

1. Check the path of Xcode command-line tools currently used on your system by running `xcode-select --print-path` in your terminal, it should display something like:
    ```
    $ xcode-select --print-path
    /Applications/Xcode.app/Contents/Developer
    ```
2. Check the version of Xcode command-line tools currently used on your system by running `xcodebuild -version` in your terminal, it should display something like:
    ```
    $ xcodebuild -version
    Xcode 15.3
    Build version 15E5202a
    ```
3. Change the default version to `Xcode_15.2.0.app` by running `sudo xcode-select -s /Applications/Xcode_15.2.0.app` in your terminal
4. Verify that the default version is now indeed 15.2:
    ```
    $ xcodebuild -version
    Xcode 15.2
    Build version 15C500b
    ```
To switch back to the current version simply do `sudo xcode-select -s /Applications/Xcode.app` and verify with the above steps that the command has succeeded.

Hope it helps.

**Reply by phillippschmedt** [reply]
Easiest way to install multiple xcodes versions is https://github.com/XcodesOrg/xcodes or with GUI https://github.com/XcodesOrg/XcodesApp
